[test] Flaky TLS system test

There is a new flaky system test, probably related with the changes introduced when switching to python3:

`test_tls.TestSSLEnabledNoClientVerificationTest.test_http_fails`

**Error Message**

Exception not raised

**Stacktrace**
```
Traceback (most recent call last):
  File "/usr/lib/python3.7/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib/python3.7/unittest/case.py", line 615, in run
    testMethod()
  File "/go/src/github.com/elastic/apm-server/tests/system/test_tls.py", line 109, in test_http_fails
    data=self.get_event_payload())
  File "/usr/lib/python3.7/unittest/case.py", line 203, in __exit__
    self._raiseFailure("{} not raised".format(exc_name))
  File "/usr/lib/python3.7/unittest/case.py", line 135, in _raiseFailure
    raise self.test_case.failureException(msg)
AssertionError: Exception not raised

```

--- a/tests/system/test_tls.py
+++ b/tests/system/test_tls.py
@@ -104,9 +104,13 @@ class TestSSLEnabledNoClientVerificationTest(TestSecureServerBaseTest):
 
     def test_http_fails(self):
         with self.assertRaises(Exception):
-            return requests.post("http://localhost:8200/intake/v2/events",
-                                 headers={'content-type': 'application/x-ndjson'},
-                                 data=self.get_event_payload())
+            with requests.Session() as session:
+                try:
+                    return session.post("http://localhost:8200/intake/v2/events",
+                                        headers={'content-type': 'application/x-ndjson'},
+                                        data=self.get_event_payload())
+                finally:
+                    session.close()
 
 
 @integration_test

